test/feature/steps/init.js

```diff
@@ -1,72 +1,129 @@
 const {
   Given, When, Then, Before, After,
 } = require('cucumber');
-const Preferences = require('preferences');
-const { exec } = require('child_process');
 const chai = require('chai');
-const sinon = require('sinon');
-const sinonChai = require('sinon-chai');
-const path = require('path');
-const figlet = require('figlet');
+const chaiAsPromised = require('chai-as-promised');
 const chalk = require('chalk');
+const fs = require('fs');
 const inquirer = require('inquirer');
+const path = require('path');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+
 const controller = require('../../../lib/controller');
 const initInput = require('../../../lib/cli/input/init');
 const preferenceManager = require('../../../lib/utils/preference-manager');
+const { logger } = require('../../../lib/utils/logger');
+
 
 chai.use(sinonChai);
+chai.use(chaiAsPromised);
 chai.should();
 
-global.promptStub = sinon.stub(inquirer, 'prompt');
-global.logSpy = sinon.stub(console, 'log');
+Before(function () {
+  this.promptStub = sinon.stub(inquirer, 'prompt');
+  this.logSpy = sinon.stub(console, 'log');
+  this.loggerStub = sinon.stub(logger, 'log');
+});
+
+Given('that the gitlab host is set from the file {string}', function (file) {
```

Comment:
This background function is still in `init.js`. This being a background step, needs to be in a separate file.